core/src/main/scala/org/apache/spark/api/java/JavaUtils.scala

```diff
@@ -89,7 +91,7 @@ private[spark] object JavaUtils {
                   mm.remove(k)
                   prev = None
                 case _ =>
-                  throw new UnsupportedOperationException("remove")
```

Comment:
How can an end-user hit this error?

Reply:
How can we identify which error is user-facing error or not? Is there any rule we can apply?

Reply:
Usually, non-user-facing errors can be identified by the exception type, for example, `IllegalStateException`, or by the code logic. Here it seems if `underlying` is not a mutable map, then the `remove` operation is not supported. This seems to be a develop-facing error.

Reply:
Thanks, so now we should determine which error is develop-facing error and restore it as before right?
What above previous pull requests that have already merged?

Reply:
> What above previous pull requests that have already merged?

Those should be revisited. It's not urgent as it's an internal refactor, and end-users won't notice it. We can enforce this in new PRs first.